src/main/java/de/tum/in/www1/artemis/config/migration/entries/MigrationEntry20230808_203400.java

```diff
@@ -0,0 +1,460 @@
+        }
+        catch (VersionControlException | ContinuousIntegrationException | NoSuchElementException | RestClientException e) {
+            log.warn("Failed to replace solution repository in template build plan for studentParticipationId {} with buildPlanId {} of exerciseId {} ",
+                    studentParticipation.getId(), studentParticipation.getBuildPlanId(), exercise.getId(), e);
+            errorMap.put(exercise, false);
+        }
+        try {
+            ciMigrationService.orElseThrow().overrideBuildPlanRepository(studentParticipation.getBuildPlanId(), "tests", exercise.getTestRepositoryUrl(), exercise.getBranch());
+        }
+        catch (VersionControlException | ContinuousIntegrationException | NoSuchElementException | RestClientException e) {
+            log.warn("Failed to replace tests repository in solution build plan for studentParticipationId {} with buildPlanId {} of exerciseId {} ", studentParticipation.getId(),
+                    studentParticipation.getBuildPlanId(), exercise.getId(), e);
+            errorMap.put(exercise, false);
+        }
+        for (var auxiliary : auxiliaryRepositories) {
+            try {
+                ciMigrationService.orElseThrow().overrideBuildPlanRepository(studentParticipation.getBuildPlanId(), auxiliary.getName(), auxiliary.getRepositoryUrl(),
+                        exercise.getBranch());
+            }
+            catch (VersionControlException | ContinuousIntegrationException | NoSuchElementException | RestClientException e) {
+                log.warn("Failed to replace {} repository in solution build plan for studentParticipationId {} with buildPlanId {} of exerciseId {} ", auxiliary.getName(),
+                        studentParticipation.getId(), studentParticipation.getBuildPlanId(), exercise.getId(), e);
+                errorMap.put(exercise, true);
+            }
+        }
+        try {
+            ciMigrationService.orElseThrow().overrideRepositoriesToCheckout(studentParticipation.getBuildPlanId(), auxiliaryRepositories);
+        }
+        catch (VersionControlException | ContinuousIntegrationException | NoSuchElementException | RestClientException e) {
+            log.warn("Failed to replace repositories in solution build plan for studentParticipationId {} with buildPlanId {} of exerciseId {} ", studentParticipation.getId(),
+                    studentParticipation.getBuildPlanId(), exercise.getId(), e);
+            errorMap.put(exercise, false);
+        }
+    }
+
+    /**
+     * Migrates a single solution build plan.
+     *
+     * @param exercise The exercise to migrate
+     * @param errorMap The error map to write errors to
+     */
+    private void migrateSolutionBuildPlan(ProgrammingExercise exercise, List<AuxiliaryRepository> auxiliaryRepositories, Map<ProgrammingExercise, Boolean> errorMap) {
+        try {
```

Comment:
Same thing here as @dfuchss mentioned with the difficulty of reading all the try-catches.